safenode/src/network/mod.rs

```diff
@@ -318,3 +326,109 @@ impl Network {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::SwarmDriver;
+    use crate::log::init_node_logging;
+    use eyre::{eyre, Result};
+    use libp2p::{
+        kad::kbucket::{Entry, InsertResult, KBucketsTable, Key, NodeStatus},
+        PeerId,
+    };
+    use rand::thread_rng;
+    use std::{
+        collections::{BTreeMap, HashMap},
+        fmt,
+        time::Duration,
+    };
+    use xor_name::XorName;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn closest() -> Result<()> {
```

Comment:
I think this doesn't quite follow the test naming guidelines.. :thinking_face:

`closest_actually_are_the_closest` isn't a perfect name, just an example from the top of my head... but I think it shall be something like that.